services/identity-service/src/did/vault.service.ts

```diff
@@ -0,0 +1,48 @@
+import { Injectable } from "@nestjs/common";
+const Vault = require('hashi-vault-js');
+
+
+
+@Injectable()
+export class VaultService {
+    constructor () {}
+    private token = process.env.VAULT_TOKEN;
+    private vault = new Vault( {
+            https: false,
+            baseUrl: `${process.env.VAULT_ADDR}/v1`,
+            rootPath: `${process.env.VAULT_ADDR}/v1/kv`,
```

Comment:
can be made configurable throughout. We can have vault address and root path as two env variables.

Reply:
added as env vars